### ISSUE TITLE
[4.x]: Cannot eager load entry fields on matrix blocks

### ISSUE THREAD

**Issue by ivanBereznev** [craftcms/cms#12631]
### Description
There is an entry type named `Product` which has an `productAttributes` field. This field contains Matrix Blocks of several types where each type has its own entry field that references entries.
When I want to query products along with all Matrix Blocks with their relations I do the following:
```
$products = (new EntryQuery(Entry::class))
  ->type('product')
  ->with(['productAttributes.typeHandleA:attributeA', 'productAttributes.typeHandleB:attributeB'])
  ->all();
```
Then I do data processing without generating too many DB queries while iterating over products and their attributes.

This worked perfectly with Craft 4.3.1 but does not work with Craft 4.3.7, although data and data model has not been changed in any ways

### Steps to reproduce

1. Have a Matrix field on an entry type and some Entry fields to these block types.
2. Create an Entry of aforementioned type, add some Matrix Blocks and entries related to them.
3. Try to load Matrix blocks and their relations eagerly like in the description above.
4. Access eager loaded fields of Matrix blocks via magic getters or via `getEagerLoadedElements` method.

### Expected behavior
Related entries are loaded eagerly and available, magic getters and `getEagerLoadedElements` return collections of elements.

### Actual behavior
Entries related to matrix blocks are not loaded, magic getters return `ElementQuery` instances. `getEagerLoadedElements` returns blank result.

### Craft CMS version

4.3.7

### PHP version

8.0.22

### Operating system and version

Debian GNU/Linux 11

### Database type and version

PostgreSQL 13.6

### Installed plugins and versions

-

**Comment by sebastian-lenz** [comment]
Actually eager loading of fields on matrix fields is completely broken, please see my comment here:
https://github.com/craftcms/cms/issues/12539#issuecomment-1418959299

**Comment by i-just** [comment]
Hi, thanks for reporting! A new PR to fix this issue has been created.

**Comment by brandonkelly** [comment]
This wasn’t actually fixed in 4.3.8 as intended, but 4.3.8.1 is out now with the proper fix.

**Comment by brandonkelly** [comment]
Note that if you updated to 4.3.8 already, you will be affected by [this bug](https://github.com/craftcms/cms/issues/12647), which was also fixed in 4.3.8.1. If so you will need to run `composer update` to fix.

**Comment by arentsen** [comment]
After updating to 3.7.64.1, I got the following error `Undefined array key 0` related to this query: `{% set artistQuery = craft.entries.site(siteHandle).section('artists').title(artistName).with(['featuredImageArtist', 'artistPhotos', 'artistDiscography.artistDiscography:discographyPhoto', 'biographyPDF', 'externalLink.externalLinks:linktoAsset', 'externalLink.externalLinks:linktoEntry']).one() %}`

When I removed the eager loading part, it still threw the error until I did the composer update. However, it still does not work when I add the parts after the colons. I had to remove "linktoAsset", "linktoEntry", "discographyPhoto" and for some reason also the entire field "biographyPDF". Am I now losing the eager loading or does craft automatically eager load any assets after the update?
